mibidata/segmentation.py

```diff
@@ -49,35 +57,55 @@ def extract_cell_dataframe(label_image, image=None, mode='total'):
             if mode == 'total':
                 vals = image.data[nonzeros[0], nonzeros[1], :].sum(axis=0)
             elif mode == 'quadrant':
-                vals = _quadrant_mean(nonzeros, image)
+                vals = _circular_sectors_mean(nonzeros, image, num_sectors=4)
+            elif mode == 'circular_sectors':
+                vals = _circular_sectors_mean(nonzeros, image, num_sectors)
             else:
-                raise ValueError('"mode" must be either "total" or "quadrant"')
+                raise ValueError('"mode" must be either "total", "quadrant" or \
+                "circular_sectors"')
             row.extend(vals)
         rows.append(row)
     return pd.DataFrame(rows, columns=columns).set_index('label')
 
 
-def _quadrant_mean(inds, image):
-    """Gets the geometric mean of a regions's intensities across its quadrants.
+def _circular_sectors_mean(inds, image, num_sectors=8):
+    """Divide a region in circular sectors and get the geometric mean across the
+    sectors.
 
     Args:
         inds: A tuple of 2 arrays of the y- and x- indices of the pixels in a
             segmented region of an image.
         image: A MibiImage in which the corresponding pixel indices are located.
+        num_sectors: number of circular sectors to use. Optional, default is 8.
 
     Returns:
         An array whose length is equal to the number of channels in the image.
         Each value in the array is the geometric mean of the image's integrated
-        channel intensities over the regions's four quadrants.
+        channel intensities over the regions's num_sectors circular sectors.
     """
+    # calculate the geometric center of the cells and get the counts
     y_center, x_center = np.mean(inds, axis=1)
     vals = image.data[inds]  # has shape (num_pixels_in_cell, num_channels)
-    upper_left = vals[(inds[0] <= y_center) & (inds[1] < x_center)].sum(axis=0)
-    upper_right = vals[(inds[0] < y_center) & (inds[1] >= x_center)].sum(axis=0)
-    lower_left = vals[(inds[0] > y_center) & (inds[1] <= x_center)].sum(axis=0)
-    lower_right = vals[(inds[0] >= y_center) & (inds[1] > x_center)].sum(axis=0)
-    quads = np.stack((upper_left, upper_right, lower_left, lower_right), axis=1)
-    return np.power(np.product(quads, axis=1), 1 / 4)
+
+    # convert to polar coordinates: y, x -> phi, r
+    phi = util.car2pol(inds[1], inds[0], x_center, y_center)[1]
+
+    # create circular sectors
+    sectors = []
+    ang_step = 2.*np.pi/num_sectors
+    for i in range(num_sectors):
+        values = vals[(phi >= i*ang_step) & (phi < (i + 1)*ang_step)]
+         # check if the sector is empty; if so, fill one (neutral element for
+         # the multiplication in the geometric mean); otherwise the whole cell
+         # will be set to zero
+        if not values.size:
+            values = np.ones((1, len(image.channels)))
```

Comment:
I have mixed feelings about this -- it was intentional in the quadrants scheme to send everything to zero if the marker was not present in all quadrants. But I understand that when there are more sectors this can bias more things to zero than we'd like --that's partly what I wanted to see in [this comment](https://github.com/ionpath/mibitracker-client/pull/37#issuecomment-501356369). I wonder if we want to make it an option to use zeros or ones here. Having it default to zero would also make this backwards-compatible with our previous usage.

Reply:
@rfinck I support @mapazarr implementation. Values are set to 1 only if the size of the sector is 0. 
However, I think it would be advisable to throw a warning whenever that happens.

Reply:
@rfink the behavior you describe from the quadrants method is kept in the current implementation: if the marker is not present in a sector it's set to zero, hence the cell is discarded. BUT, if the sector is empty (meaning there are no pixels), there is no information if the marker is present or not. In the old quadrants implementation a zero would be set for the sector, discarding the whole cell, although we don't have enough information for discarding it, since the sector is empty. My implementation sets a one in order to let the sectors that contain pixels decide if the cell is kept or not. Please let me know if you need more clarification or if you want me to change something in the code. Throwing a warning as @stanislaw-ionpath suggests might be a good idea. Let me know if you agree.

Reply:
Thanks for clarifying my misunderstanding. This sounds fine as it is now.